spring-ai-core/src/main/java/org/springframework/ai/model/function/FunctionCallback.java

```diff
@@ -116,63 +116,67 @@ enum SchemaType {
 	 * <li>{@link MethodInvokingSpec} - The method invoking builder interface.
 	 * </ul>
 	 */
-	interface Builder {
+	interface Builder extends CommonInvokingSpec<Builder> {
```

Comment:
The Builder extends the CommonInvokingSpec<Builder> only for backward compatibility. 
But this might cause confusion and requires merging logic inside the function() and method() leafs. 
Perhaps we can remove it all together causing a breaking change?